### ISSUE TITLE
Bitmap length is 17 if max_length for primary and secondary bitmap is 16

### ISSUE THREAD

**Issue by mumin91** [knovichikhin/pyiso8583#8]
This is my spec:
```
default_ascii = {
"h":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 8,   "desc": "Message Header"},
"t":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4,   "desc": "Message Type"},
"p":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "Bitmap, Primary"},
"1":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "Bitmap, Secondary"},
"2":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 19,  "desc": "Primary Account Number (PAN)"},
"3":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 7,   "desc": "Processing Code"},
"4":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 12,  "desc": "Amount, Transaction"},
"5":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 12,  "desc": "Amount, Settlement"},
"6":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 12,  "desc": "Amount, Cardholder Billing"},
"7":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 10,  "desc": "Transmission Date and Time"},
"8":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 8,   "desc": "Amount, Cardholder Billing Fee"},
"9":   {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 8,   "desc": "Conversion Rate, Settlement"},
"10":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 8,   "desc": "Conversion Rate, Cardholder Billing"},
"11":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 6,   "desc": "System Trace Audit Number"},
"12":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 6,   "desc": "Time, Local Transaction"},
"13":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4,   "desc": "Date, Local Transaction"},
"14":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4,   "desc": "Date, Expiration"},
"15":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4,   "desc": "Date, Settlement"},
"16":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4,   "desc": "Date, Conversion"},
"17":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4,   "desc": "Date, Capture"},
"18":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4,   "desc": "Merchant Type or Standard Industrial Classification (SIC) or Merchant Category Code (MCC)"},
"19":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Acquiring Institution Country Code"},
"20":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "PAN Country Code"},
"21":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Forwarding Institution Country Code"},
"22":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Point-of-Service Entry Mode"},
"23":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "PAN Sequence Number / MBR"},
"24":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Network International ID (NII)"},
"25":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Point-of-Service Condition Code"},
"26":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4,   "desc": "Message Reason Code"},
"27":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 1,   "desc": "Authorizing ID Response Length"},
"28":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 9,   "desc": "Amount, Transaction Fee"},
"29":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 9,   "desc": "Amount, Settlement Fee"},
"30":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 9,   "desc": "Amount, Transaction Processing Fee"},
"31":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 9,   "desc": "Amount, Settlement Processing Fee"},
"32":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 11,  "desc": "Acquiring Institution ID Code"},
"33":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 11,  "desc": "Forwarding Institution ID Code"},
"34":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 28,  "desc": "Primary Account Number, Extended"},
"35":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 37,  "desc": "Track 2 Data"},
"36":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 104, "desc": "Track 3 Data"},
"37":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 12,  "desc": "Retrieval Reference Number"},
"38":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 6,   "desc": "Authorization ID Response"},
"39":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 5,   "desc": "Response Code"},
"40":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Service Restriction Code"},
"41":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 16,   "desc": "Card Acceptor Terminal ID"},
"42":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 15,  "desc": "Card Acceptor ID Code"},
"43":  {"len_count": "nibbles", "data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 222,  "desc": "Card Acceptor Name/Location"}, # getting error here
"44":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 25,  "desc": "Additional Response Data"},
"45":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 76,  "desc": "Track 1 Data"},
"46":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Additional Data - ISO"},
"47":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Additional Data - National"},
"48":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 31, "desc": "Additional Data - Private/Reference to Other Transaction"},
"49":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Currency Code, Transaction"},
"50":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Currency Code, Settlement"},
"51":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Currency Code, Cardholder Billing"},
"52":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "PIN"},
"53":  {"data_enc": "ascii", "len_enc": "b", "len_type": 0, "max_len": 48,  "desc": "Security-Related Control Information"},
"54":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 12, "desc": "Adjustment Amounts"},
"55":  {"data_enc": "ascii", "len_enc": "b", "len_type": 3, "max_len": 255, "desc": "ICC data"},
"56":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved ISO"},
"57":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved National"},
"58":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved National"},
"59":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved National"},
"60":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved National"},
"61":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 14, "desc": "Card Issuer Data"},
"62":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "External Transaction Attributes"},
"63":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16, "desc": "New Pin"},
"64":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "MAC"},
"65":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "Bitmap, Extended"},
"66":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 1,   "desc": "Settlement Code"},
"67":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 2,   "desc": "Extended Payment Code"},
"68":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Receiving Institution Country Code"},
"69":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Settlement Institution Country Code"},
"70":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 3,   "desc": "Network Management Information Code"},
"71":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4,   "desc": "Message Number"},
"72":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4,   "desc": "Message Number, Last"},
"73":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 6,   "desc": "Date, Action"},
"74":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 10,  "desc": "Credits, Number"},
"75":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 10,  "desc": "Credits, Reversal Number"},
"76":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 10,  "desc": "Debits, Number"},
"77":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 10,  "desc": "Debits, Reversal Number"},
"78":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 10,  "desc": "Transfer, Number"},
"79":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 10,  "desc": "Transfer, Reversal Number"},
"80":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 10,  "desc": "Inquiries, Number"},
"81":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 10,  "desc": "Authorizations, Number"},
"82":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 12,  "desc": "Credits, Processing Fee Amount"},
"83":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 12,  "desc": "Credits, Transaction Fee Amount"},
"84":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 12,  "desc": "Debits, Processing Fee Amount"},
"85":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 12,  "desc": "Debits, Transaction Fee Amount"},
"86":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "Credits, Amount"},
"87":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "Credits, Reversal Amount"},
"88":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "Debits, Amount"},
"89":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "Debits, Reversal Amount"},
"90":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 42,  "desc": "Original Data Elements"},
"91":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 1,   "desc": "File Update Code"},
"92":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 2,   "desc": "File Security Code"},
"93":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 5,   "desc": "Response Indicator"},
"94":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 7,   "desc": "Service Indicator"},
"95":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 24,  "desc": "Replacement Amounts"},
"96":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "Message Security Code"},
"97":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 17,  "desc": "Amount, Net Settlement"},
"98":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 25,  "desc": "Payee"},
"99":  {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 11,  "desc": "Settlement Institution ID Code"},
"100": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 11,  "desc": "Receiving Institution ID Code"},
"101": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 17,  "desc": "File Name"},
"102": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 30,  "desc": "Account ID 1 (FROM)"},
"103": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 30,  "desc": "Account ID (T0)"},
"104": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 4, "desc": "Host Net Identification"},
"105": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 25, "desc": "Account Balance Data"},
"106": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 46, "desc": "Multi Currency Data"},
"107": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 12, "desc": "Final RRN"},
"108": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Regional Listing Data/String Message"},
"109": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Multi Account Data"},
"110": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 1, "max_len": 9, "desc": "Numeric Message"},
"111": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 99, "desc": "Account Identification 2"},
"112": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved for National Use"},
"113": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved for National Use"},
"114": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 250, "desc": "Mini-Statement Data"},
"115": {"data_enc": "ascii", "len_enc": "b", "len_type": 2, "max_len": 99999, "desc": "Statement Data"},
"116": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 99999, "desc": "Billing Data"}, # Need to verify if it is ok
"117": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved for National Use"},
"118": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved for National Use"},
"119": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved for National Use"},
"120": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Reserved for Private Use"},
"121": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 47, "desc": "Additional POS Data"},
"122": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 48, "desc": "3D-Secure Data"},
"123": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 999, "desc": "Miscellaneous Transaction Attributes"},
"124": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 99999, "desc": "Miscellaneous Transaction Attributes 2"},
"125": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 2, "max_len": 99999, "desc": "Administrative Transaction Data"},
"126": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 34, "desc": "Preauthorization Parameters"},
"127": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 3, "max_len": 99999, "desc": "Additional Information"},
"128": {"data_enc": "ascii", "len_enc": "ascii", "len_type": 0, "max_len": 16,  "desc": "MAC"}
}
```
I am getting this for primary bitmap which should be a HEX 16:   b'B238641108A08000' . The HEX of the primary bitmap is 17 characters which should be 16 characters as per my spec. Could anyone please let me know, where is the issue?

**Comment by knovichikhin** [comment]
Here is correct primary bitmap with a simple message using your spec. If you can provide the message you are trying to encode that does not work, I can take a look.

```
import iso8583
empty_message = {"h": "headerXX", "t": "9999"}
raw_msg, encoded_msg = iso8583.encode(empty_message, default_ascii)
print(raw_msg)
print("Primary bitmap length:", len(encoded_msg["p"]["data"]))
print("Primary bitmap:", encoded_msg["p"]["data"])
```

Output:
```
bytearray(b'headerXX99990000000000000000')
Primary bitmap length: 16
Primary bitmap: b'0000000000000000'
```

**Comment by mumin91** [comment]
> Here is correct primary bitmap with a simple message using your spec. If you can provide the message you are trying to encode that does not work, I can take a look.
> 
> ```
> import iso8583
> empty_message = {"h": "headerXX", "t": "9999"}
> raw_msg, encoded_msg = iso8583.encode(empty_message, default_ascii)
> print(raw_msg)
> print("Primary bitmap length:", len(encoded_msg["p"]["data"]))
> print("Primary bitmap:", encoded_msg["p"]["data"])
> ```
> 
> Output:
> 
> ```
> bytearray(b'headerXX99990000000000000000')
> Primary bitmap length: 16
> Primary bitmap: b'0000000000000000'
> ```

**Comment by knovichikhin** [comment]
It's a bit harder to help with decoding because you have to have correct spec. Without that things break in very confusing ways.
So, I'll show you how to troubleshoot and maybe it gets you closer to parsing this.

When we try parsing your line:

```
import iso8583
import pprint
raw_msg = b"A4M040000200B238641108A0800000000000048000A00100000000001500000063010313535328216314106306011050002D00000000110000000000300004625397608ABCUT552UCBL GAZIPUR 000050BONGI BRANCH RECYCLER 789 DHAKA [D] 00120180630UCBL ACBL 000000000001000005001?0000000000000000000000000001 026CC=0TC=2TL=1TLC=enTZ=1"
try:
    decoded_msg, encoded_msg = iso8583.decode(raw_msg, spec=default_ascii)
except iso8583.DecodeError as e:
    print(e)
```
The library throws an exception.

```
Field data is 11 bytes, expecting 47: field 121 pos 284
```

Unless you know spec very well that error is not very useful. I don't know your spec at all and that error does not mean much too me except that it's probably field 121 or before that. Lucky, `DecodeError` exception has a bunch of things attached to it. It can show how the message was parsed up to point of failure. Let's try again:

```
import iso8583
import pprint
raw_msg = b"A4M040000200B238641108A0800000000000048000A00100000000001500000063010313535328216314106306011050002D00000000110000000000300004625397608ABCUT552UCBL GAZIPUR 000050BONGI BRANCH RECYCLER 789 DHAKA [D] 00120180630UCBL ACBL 000000000001000005001?0000000000000000000000000001 026CC=0TC=2TL=1TLC=enTZ=1"
try:
    decoded_msg, encoded_msg = iso8583.decode(raw_msg, spec=default_ascii)
except iso8583.DecodeError as e:
    print(e)
    pprint.pp(e.doc_dec)
    pprint.pp(e.doc_enc)
```

This time we print out partially parsed document from `doc_dec` and `doc_enc`. `doc_enc` is of particular interested when trying to troubleshoot because it will show exact message breakdown.

```
Field data is 11 bytes, expecting 47: field 121 pos 284
{'h': 'A4M04000',
 't': '0200',
 'p': 'B238641108A08000',
 '1': '00000000048000A0',
 '3': '0100000',
 '4': '000001500000',
 '7': '0630103135',
 '11': '353282',
 '12': '163141',
 '13': '0630',
 '18': '6011',
 '19': '050',
 '22': '002',
 '28': 'D00000000',
 '32': '00000000003',
 '37': '000046253976',
 '41': 'ABCUT552',
 '43': 'UCBL GAZIPUR 000050BONGI BRANCH RECYCLER 789 DHAKA [D] 00120180630UCBL '
       'ACBL 000000000001000005001?0000000000000',
 '49': '000',
 '102': '',
 '105': '0000000001 026CC=0TC=2TL=',
 '121': ''}
{'h': {'len': b'', 'data': b'A4M04000'},
 't': {'len': b'', 'data': b'0200'},
 'p': {'len': b'', 'data': b'B238641108A08000'},
 '1': {'len': b'', 'data': b'00000000048000A0'},
 '3': {'len': b'', 'data': b'0100000'},
 '4': {'len': b'', 'data': b'000001500000'},
 '7': {'len': b'', 'data': b'0630103135'},
 '11': {'len': b'', 'data': b'353282'},
 '12': {'len': b'', 'data': b'163141'},
 '13': {'len': b'', 'data': b'0630'},
 '18': {'len': b'', 'data': b'6011'},
 '19': {'len': b'', 'data': b'050'},
 '22': {'len': b'', 'data': b'002'},
 '28': {'len': b'', 'data': b'D00000000'},
 '32': {'len': b'11', 'data': b'00000000003'},
 '37': {'len': b'', 'data': b'000046253976'},
 '41': {'len': b'08', 'data': b'ABCUT552'},
 '43': {'len': b'',
        'data': b'UCBL GAZIPUR 000050BONGI BRANCH RECYCLER 789 DHAKA [D] 00120'
                b'180630UCBL ACBL 000000000001000005001?0000000000000'},
 '49': {'len': b'', 'data': b'000'},
 '102': {'len': b'00', 'data': b''},
 '105': {'len': b'', 'data': b'0000000001 026CC=0TC=2TL='},
 '121': {'len': b'', 'data': b'1TLC=enTZ=1'}}
```

So, by just looking at this I would say field `105` looks suspicious. I think the spec for it is too long or the data coming in is too short. Because that `026` looks be a length of an LLLVAR field.

I hope this helps.

**Comment by mumin91** [comment]
I am looking into it. Thank you so much for your effort.

**Comment by mumin91** [comment]
My question is about the primary bitmap and secondary bitmaps. It shows as below:
```
'p': 'B238641108A08000',
 '1': '00000000048000A0',
```
But if you look in the specification, you will see that I have used max_len 16 for those fields. But It takes 17 characters for those fields which will create problem in decoding the values. I have also looked into `_decode_bitmaps` method to find the issue. :) I think you got my point.
